src/include/utils/backend_status.h

```diff
@@ -169,9 +169,11 @@ typedef struct PgBackendStatus
 	/* query identifier, optionally computed using post_parse_analyze_hook */
 	uint64		st_query_id;
 
-	int64		st_active_time;
-	int64 		st_transaction_idle_time;
-	int64 		st_idle_time;
+	/* time spent in respective states in milliseconds */
+	float8		st_total_active_time;
+	float8 		st_total_transaction_idle_time;
+	float8 		st_total_idle_time;
```

Comment:
IMO it should be double, at least this is what is used in pgss for time counters: https://github.com/sdudoladov/postgres/blob/73dc9f8a1c2d36604708515c1250f952b00411ae/contrib/pg_stat_statements/pg_stat_statements.c#L162-L192

Reply:
Or maybe it should be PgStat_Counter, and recalculation (/1000) happen in the pg_stat_get_activity()